etl/churn.kp/orig_src/Churn.ipynb

```diff
@@ -588,29 +614,56 @@
     "                x.get('sync_usage', 'unknown'),\n",
     "                x.get('current_version', 'unknown'),\n",
     "                x.get('current_week', 'unknown'),\n",
-    "                x.get('is_active', 'unknown')), (1, x.get('usage_hours', 0), x.get('squared_usage_hours', 0))))\n",
+    "                x.get('source', 'unknown'),\n",
+    "                x.get('medium', 'unknown'),\n",
+    "                x.get('campaign', 'unknown'),\n",
+    "                x.get('content', 'unknown'),\n",
+    "                x.get('distribution_id', 'unknown'),\n",
+    "                x.get('default_search_engine', 'unknown'),\n",
+    "                x.get('locale', 'unknown'),\n",
+    "                x.get('is_active', 'unknown')\n",
+    "            ),(\n",
+    "                1,  # active users \n",
+    "                x.get('usage_hours', 0),\n",
+    "                x.get('squared_usage_hours',0) \n",
+    "            )\n",
+    "        )\n",
+    "    )\n",
     "\n",
     "    def reduce_func(x, y):\n",
     "        return (x[0] + y[0], # Sum active users\n",
     "                x[1] + y[1], # Sum usage_hours\n",
     "                x[2] + y[2]) # Sum squared_usage_hours\n",
+    "    \n",
     "    aggregated = countable.reduceByKey(reduce_func)\n",
     "    \n",
     "    records_df = aggregated.map(lambda x: x[0] + x[1]).toDF(record_columns)\n",
     "\n",
     "    if enable_upload_csv:\n",
     "        churn_outfile = get_churn_filename(week_start, week_end)\n",
+    "        # groupby and aggregate over the original attributes\n",
```

Comment:
The new attributes being used in bug 1323598 are going to be read from the s3://telemetry-parquet instead of the csv file, so `enable_upload_csv` should only be there for the jobs that depend on it.